storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -91,18 +90,14 @@ public class RemoteIndexCache implements Closeable {
      */
     private final AtomicBoolean isRemoteIndexCacheClosed = new AtomicBoolean(false);
 
-    /**
-     * Unbounded queue containing the removed entries from the cache which are waiting to be garbage collected.
-     */
-    private final LinkedBlockingQueue<Entry> expiredIndexes = new LinkedBlockingQueue<>();
-
     /**
      * Lock used to synchronize close with other read operations. This ensures that when we close, we don't have any other
      * concurrent reads in-progress.
      */
-    private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
     private final RemoteStorageManager remoteStorageManager;
-    private final ShutdownableThread cleanerThread;
+    private final KafkaScheduler cleanerScheduler = new KafkaScheduler(1, true, REMOTE_LOG_INDEX_CACHE_CLEANER_THREAD);
+    private int fileDeleteDelayMs = 60_000;
```

Comment:
I think the value of 60 secs looks to be reasonable. We can revisit setting a config if really needed in future.